cg/meta/upload/gt.py

```diff
@@ -43,93 +47,168 @@ def data(self, analysis: Analysis) -> dict:
             }
         return samples_sex
 
-    def _get_samples_sex_balsamic(self, case_obj: Case) -> dict:
-        samples_sex = {}
-        for link_obj in case_obj.links:
-            if link_obj.sample.is_tumour:
-                continue
-            sample_id = link_obj.sample.internal_id
+    def _get_samples_sex_mip_dna(self, case: Case) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction"""
+        qc_metrics_file: Path = self._get_qcmetrics_file(case_id=case.internal_id)
+        analysis_sex: dict = self._get_analysis_sex_mip_dna(qc_metrics_file)
+        samples_sex: dict[str, dict[str, str]] = {}
+        for sample in case.samples:
+            sample_id: str = sample.internal_id
             samples_sex[sample_id] = {
-                "pedigree": link_obj.sample.sex,
-                "analysis": Sex.UNKNOWN,
+                "pedigree": sample.sex,
+                "analysis": analysis_sex[sample_id],
             }
         return samples_sex
 
-    def analysis_sex(self, qc_metrics_file: Path) -> dict:
-        """Fetch analysis sex for each sample of an analysis."""
-        qc_metrics: MIPMetricsDeliverables = self.get_parsed_qc_metrics_data(qc_metrics_file)
+    @staticmethod
+    def _get_analysis_sex_mip_dna(qc_metrics_file: Path) -> dict:
+        """Return analysis sex for each sample of an analysis."""
+        qc_metrics: MIPMetricsDeliverables = (
+            UploadGenotypesAPI._get_parsed_qc_metrics_deliverables_mip_dna(qc_metrics_file)
+        )
         return {
             sample_id_metric.sample_id: sample_id_metric.predicted_sex
             for sample_id_metric in qc_metrics.sample_id_metrics
         }
 
-    def get_bcf_file(self, hk_version_obj: Version) -> File:
-        """Fetch a bcf file and return the file object"""
-        genotype_files: list = self._get_genotype_files(version_id=hk_version_obj.id)
-        for genotype_file in genotype_files:
-            if self._is_variant_file(genotype_file=genotype_file):
-                LOG.debug(f"Found bcf file {genotype_file.full_path}")
-                return genotype_file
-        raise FileNotFoundError(f"No vcf or bcf file found for bundle {hk_version_obj.bundle_id}")
-
-    def get_qcmetrics_file(self, hk_version_obj: Version) -> Path:
-        """Fetch a qc_metrics file and return the path"""
-        hk_qcmetrics = self.hk.files(
-            version=hk_version_obj.id, tags=HkMipAnalysisTag.QC_METRICS
-        ).first()
-        LOG.debug(f"Found qc metrics file {hk_qcmetrics.full_path}")
+    def _get_bcf_file(
+        self,
+        case_id: str,
+    ) -> File:
+        """Return a Housekeeper file object.
+        Raises: FileNotFoundError if nothing is found in the Housekeeper bundle."""
+        LOG.debug("Get Genotype files from Housekeeper")
+        genotype_file: File = self._get_genotype_file(case_id)
+        if self._is_variant_file(genotype_file):
```

Comment:
This check already seems to be done in self._get_genotype_file now

Reply:
Yes. Removed